### ISSUE TITLE
How to trigger

### ISSUE THREAD

**Issue by chrisegge** [namgk/node-red-contrib-pythonshell#12]
How do you trigger the node to start running automatically

**Comment by namgk** [comment]
Hi you can wire an inject node to it.

**Comment by chrisegge** [comment]
Yea I had the continuous checked so it wouldn't let me. What about the output? Instead of individual outputs like a function node running it's an ever growing array of data. Is this normal?

**Comment by namgk** [comment]
Can you share a sample flow?

**Comment by DotoPototo** [comment]
When ticking the 'continuous' option, you can no longer inject into the pythonshell node. This means I can't get the pythonshell node to start on startup

**Comment by namgk** [comment]
As I remember, it starts when it gets the first message. Can you double check?

**Comment by DotoPototo** [comment]
I tried but it did not work. The only way I can get it working is to manually start it.

**Comment by ozett** [comment]
yeah, still buggy... cant check "continous" or "std-in" separately.... hope the update is integrated soon -> https://github.com/namgk/node-red-contrib-pythonshell/issues/15#issuecomment-663958044

**Comment by BigJerBD** [comment]
@ozett  the node , when continuous is true , by default  receives input with stdin. This is why once you set have continuous  to True you cannot  check the stdin button anymore.  It is also because a continuous  that does not recieve any input does not make a lot of sense . 

A process that have stdIn to true without being continuous will be run once when a payload is recieved. If you send an other payload, it will not trigger another time,  I rarely use in and just use continuous instead most of the time : 

* If I want python to use the same process I just use a while loop in my python code  get the inputs
* if not , once the python process is over you can still send an other payload to the node, it will start another process.

Would you mind explaining your use case? Maybe I could recommend you the proper way to run your node depending on your answer

**Comment by ozett** [comment]
my use case:
my camera at the front door triggers internally motion-detection, i pass this motion-alert to mqtt in node-red, grab an image and feed this image into an python script for face-detection.

to be a little flexible i wanted to pass some parameters via std-in from node-red to the python script.
i thought i have to check the box named "std-in" to pass my args to the script.
but that did not worked.

i only wanted to trigger the script once for every image, but i sounds a cool option to let it run contiunously.
i dont see clearly how to configure this. but could be cool maybe to feed a video-sequence to the script.

![image](https://user-images.githubusercontent.com/5937811/88666357-40678800-d0e0-11ea-8612-9ea2b75c75d2.png)

**Comment by ozett** [comment]
i assumed that i had to check th "std-in" box, to send my args via std-in from node-red to the script.
but std-in alone is not checkable...
(i am also not sure if the virtual-path is correct and works as i expect.
normally i switched to the virtual-env on the  cmd-line with: "workon virtname"...

![pynode](https://user-images.githubusercontent.com/5937811/88668953-938f0a00-d0e3-11ea-8ed0-bfc43748937a.gif)

**Comment by BigJerBD** [comment]
In your situation I think  setting continuous and stdin to True will work.   The word continuous does not mean that the script is alway running, but only that the node can receive  a stream of input

I have similar use case when instead of image input , i have text data that needs a python trigger . 
I those cases I set both to true and in my code I call the `input()` python  function to get the data from stdin 

the node will then : 
- receive the payload 
- start the python script (without any argument ) 
- pipe the payload into the process (thats why you need `input()` in your code
- process ends

if an other image is received it should do the same thing again and again

**Comment by ozett** [comment]
i will give it a try.
any idea how to activate my virtual environment from the node?
i normally use `workon `to activate -> https://virtualenvwrapper.readthedocs.io/en/latest/command_ref.html#workon

**Comment by ozett** [comment]
found some hints to "activate" virt-envs while using python3 directly from the path of the virt-env.
https://stackoverflow.com/a/26172104

what is the way with this pythonshell-node? any example available? thx.
